packages/react/src/hooks/usePreventDuplicateClick.ts

```diff
@@ -0,0 +1,20 @@
+import { useCallback, useRef, useState } from 'react';
+
+export const usePreventDuplicateClick = () => {
+  const [disabled, setDisabled] = useState(false);
+  const loadingRef = useRef(false);
+
+  const handleClick = useCallback(async (callback: () => Promise<void>) => {
+    if (loadingRef.current) return;
+
+    loadingRef.current = true;
+    setDisabled(true);
+
+    await callback();
+
+    loadingRef.current = false;
+    setDisabled(false);
+  }, []);
+
+  return { disabled, handleClick };
+};
```

Comment:
p4) 유틸리티의 관점에서는 `usePreventDuplicateClick` 보다는 `usePromise` 가 더 낫지 않나 싶네요.

```suggestion
export const usePromise = () => {
  const [isPending, setIsPending] = useState(false);

  const callPromise = useCallback(async (callback: () => Promise<void>) => {
    setIsPending(true);
    await callback();
    setIsPending(false);
  }, []);

  return { isPending, callPromise };
};
```

Reply:
지금 와서 생각난거지만 tanstack-query의 useMutation훅을 이용하면 pending 상태를 감지할 수 있으니까 굳이 이 훅이 필요할까 싶네요.